ferrocene/doc/evaluation-report/src/self-test/tool-analysis.rst

```diff
@@ -0,0 +1,104 @@
+   | **Detectable:** YES
+
+   Tool does not perform checks for some installed targets.
+
+.. hazop-error::
+   :id: ERR_SELF_TEST_FALSE_NEGATIVES
+   :caused_by: USE_SELF_TEST_CHECK
+
+   | **Risk:** Invalid output
+   | **Detectable:** NO
+
+   Tool fails to detect faulty toolchain installation.
+
+.. hazop-error::
+   :id: ERR_SELF_TEST_FALSE_POSITIVES
+   :caused_by: USE_SELF_TEST_CHECK
+
+   | **Risk:** Invalid output
+   | **Detectable:** YES
+
+   Tool falsely indicates a faulty toolchain installation.
+
+.. hazop-error::
+   :id: ERR_SELF_TEST_MODIFIES_TOOLCHAIN
+   :caused_by: USE_SELF_TEST_CHECK
+
+   | **Risk:** Invalid output
+   | **Detectable:** NO
+
+   Tool creates, changes, or deletes toolchain files/directories.
+
+.. hazop-error::
+   :id: ERR_SELF_TEST_VERSION_CHECK
+   :caused_by: USE_SELF_TEST_CHECK
+
+   | **Risk:** Invalid output
+   | **Detectable:** NO
+
+   Tool fails to detect that its toolchain is the wrong version.
+
+.. _self-test_avd:
+
+Detection Measures and Usage Restriction
+----------------------------------------
+
+Detection measures and usage restrictions are documented in the :doc:`safety
+manual <safety-manual:self-test/constraints>`.
+
+.. _ferrocene_self_test_iso_tool_classification:
+
+ISO 26262 Tool Classification
+-----------------------------
+
+During this analysis, we highlighted some of the potential errors concerning
+``ferrocene-self-test`` that impacts the safety-related software code.
+Hence, the tool impact is **TI2**.
+
+Moreover, this analysis shows us that the likelihood of detecting some of the
+potential errors is very low.
+Therefore, the tool error detection class is **TD3**.
+
+Using clause 11.4.5.4 in part 8 of the [|iso_ref|] standard, we can conclude
+that in the worst case the Tool Classification Level is **TCL3** up to ASIL D.
+Therefore, we choose the following qualification methods:
+
+- **1b.** Evaluation of the tool development process in accordance with 11.4.8
+- **1c.** Validation of the software tool in accordance with 11.4.9
+
+According to clause 11.4.2 in part 8 of the [|iso_ref|] standard,
+this choice depends on the user's software development life-cycle and their validation strategy.
+The user has the responsibility to determine whether this level, or a better one, is applicable.
+
+.. _ferrocene_self_test_iec_tool_classification:
+
+IEC 61508 Tool Classification
+-----------------------------
+
+``ferrocene-self-test`` verifies the installed toolchain,
+and is therefore an industrial [|iec_ref|] class T2 tool.
+
+IEC 62304 Tool Classification
+-----------------------------
+
+[|iec_med_ref|] does not provide an own scheme to classify and qualify tools used in its context,
+but recommends the application of techniques and tools as defined in [|iec_ref|].
+Therefore,
+with the qualification of ``ferrocene-self-test`` adhering to an IEC 61508 Tool Classification,
+``ferrocene-self-test`` can be used in the development of medical device software up to Class C.
```

Comment:
Maybe we can make this a partial as well and use it in rustc, rustfmt and ferrocene-self-test?
```suggestion
IEC 62304 Tool Classification
-----------------------------

[|iec_med_ref|] does not provide an own scheme to classify and qualify tools used in its context,
but recommends the application of techniques and tools as defined in [|iec_ref|].
Therefore,
with the qualification of the tool adhering to an IEC 61508 Tool Classification,
the tool can be used in the development of medical device software up to Class C.
```

Reply:
74fc8d31ea9